server/src/main/java/com/linecorp/centraldogma/server/internal/api/TokenService.java

```diff
@@ -192,6 +192,31 @@ public CompletableFuture<Token> updateToken(ServiceRequestContext ctx,
         );
     }
 
+    /**
+     * PATCH /tokens/{appId}/permission
+     *
+     * <p>Updates a permission of a token of the specified ID.
+     */
+    @Patch("/tokens/{appId}/level")
+    @Consumes("application/json-patch+json")
+    @RequiresAdministrator
+    public CompletableFuture<Token> updateTokenLevel(ServiceRequestContext ctx,
+                                                     @Param String appId,
+                                                     Author author, User loginUser) {
+        return getTokenOrRespondForbidden(ctx, appId, loginUser).thenCompose(
+                token -> {
+                    if (token.isAdmin()) {
+                        return mds.updateTokenToUser(author, appId).thenApply(
+                                unused -> mds.findTokenByAppId(appId).join().withoutSecret()
```

Comment:
`join()` should not be called for asynchronous code. `thenCompose()` could be used instead.

Reply:
I remove the` join()` .